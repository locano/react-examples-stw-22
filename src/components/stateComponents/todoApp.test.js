/**
 * @jest-environment jsdom
 */
/* eslint-disable no-undef */
import React from 'react';
import { render, screen, cleanup } from '@testing-library/react';
import TodoApp from './todoApp';

describe('TodoApp', () => {
  it('Render Component', () => {
    render(<TodoApp />);
    const TodoElement = screen.getByTestId('input-test');
    expect(TodoElement).toBeInTheDocument();
  });

  it('Render Component Button', () => {
    render(<TodoApp />);
    const TodoElement = screen.getByTestId('button-test');
    expect(TodoElement).toBeInTheDocument();
  });

  it('Render Component Label', () => {
    render(<TodoApp />);
    const TodoElement = screen.getByTestId('label-test');
    expect(TodoElement).toBeInTheDocument();
  });

  it('Snapshot', () => {
    const newItem = {
      text: 'new task 1',
      date: Date.now(),
    };
    const tasks = [newItem];
    const { asFragment } = render(<TodoApp tasks={tasks} />);
    expect(asFragment()).toMatchSnapshot();
  });
});
