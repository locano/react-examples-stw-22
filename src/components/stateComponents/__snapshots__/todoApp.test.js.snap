// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TodoApp Snapshot 1`] = `
<DocumentFragment>
  <h3>
    Add new Task: 
  </h3>
  <form>
    <label
      data-testid="label-test"
    >
      What do you need?
    </label>
    <input
      data-testid="input-test"
      id="todo-input"
      value=""
    />
    <button
      data-testid="button-test"
    >
       Add new Task # 2
    </button>
  </form>
  <ul>
    <li>
      new task
    </li>
  </ul>
</DocumentFragment>
`;
